File: SourceCode/Python/Binary/BinNode.py
# /* *** ODSATag: BinNode *** */
class BinNode: # Binary tree node ADT
  # Get and set the element value
  def value(self): ...
  def setValue(self, value): ...

  # Return the children
  def left(self): ... # returns a BinNode
  def right(self): ... # returns a BinNode

  # return True if a leaf node, False otherwise
  def isLeaf(self): ...
# /* *** ODSAendTag: BinNode *** */

File: SourceCode/Java_Generic/Binary/BinNode.java
/* *** ODSATag: BinNode *** */
interface BinNode<E> {
    // Binary tree node ADT

    public E value();           // Get and set the element value.
    public void setValue(E v);  // Set the element value.

    public BinNode<E> left();   // Return the left child.
    public BinNode<E> right();  // Return the right child.

    public boolean isLeaf();    // return TRUE if a leaf node, FALSE otherwise.
}
/* *** ODSAendTag: BinNode *** */
